target/linux/realtek/files-6.12/drivers/net/phy/rtl83xx-phy.c

```diff
@@ -721,41 +727,43 @@ static int rtl8380_configure_int_rtl8218b(struct phy_device *phydev)
 	msleep(100);
 
 	/* Ready PHY for patch */
-	for (int p = 0; p < 8; p++) {
-		phy_package_port_write_paged(phydev, p, RTL838X_PAGE_RAW, RTL8XXX_PAGE_SELECT, RTL821X_PAGE_PATCH);
-		phy_package_port_write_paged(phydev, p, RTL838X_PAGE_RAW, 0x10, 0x0010);
+	for (int port = 0; port < 8; port++) {
+		patchphy = get_package_phy(phydev, port);
+		phy_write_paged(patchphy, RTL838X_PAGE_RAW, RTL8XXX_PAGE_SELECT, RTL821X_PAGE_PATCH);
+		phy_write_paged(patchphy, RTL838X_PAGE_RAW, 0x10, 0x0010);
 	}
 	msleep(500);
-	for (int p = 0; p < 8; p++) {
+	for (int port = 0; port < 8; port++) {
 		int i;
 
+		patchphy = get_package_phy(phydev, port);
 		for (i = 0; i < 100 ; i++) {
-			val = phy_package_port_read_paged(phydev, p, RTL821X_PAGE_STATE, 0x10);
+			val = phy_read_paged(patchphy, RTL821X_PAGE_STATE, 0x10);
 			if (val & 0x40)
 				break;
 		}
 		if (i >= 100) {
-			phydev_err(phydev,
-			           "ERROR: Port %d not ready for patch.\n",
-			           mac + p);
+			phydev_err(patchphy, "not ready for patch.\n");
```

Comment:
I think it'd still be nice here to keep the Port # in the error message... otherwise an error is going to require kernel hacking to work out which port to further investigate.